src/agent/file_utils.cc

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "file_utils.h"
```

Comment:
Just a thought: the interlacing of files that support tests only and the files that support the agent itself feels problematic.  You do capture them in the Makefile, but it isn't intuitive.  How much would it mess things up if you created a test_utils directory?

Reply:
Yeah, fair point, I had at one point thought it may have been good to just create an entirely separate test directory for all tests and utils supporting tests, this may be a nice compromise that's isn't too bad to switch too. I'll look at doing that in its own PR.